File: src/main/java/javabot/operations/StatsOperation.java
package javabot.operations;

import java.util.ArrayList;
import java.util.List;

import javabot.BotEvent;
import javabot.Message;
import javabot.Javabot;
import javabot.dao.FactoidDao;
import org.springframework.beans.factory.annotation.Autowired;

public class StatsOperation extends BotOperation {
    @Autowired
    private FactoidDao factoidDao;

    public StatsOperation(Javabot javabot) {
        super(javabot);
    }

    private static final long startTime = System.currentTimeMillis();
    private int numberOfMessages = 0;

    @Override
    public List<Message> handleMessage(BotEvent event) {
        numberOfMessages++;
        List<Message> messages = new ArrayList<Message>();
        String message = event.getMessage();
        if (message.toLowerCase().startsWith("stats")) {
            long uptime = System.currentTimeMillis() - startTime;
            long days = uptime / 86400000;
            messages.add(
                new Message(event.getChannel(), event, "I have been up for " + days + " days, have served "
                    + numberOfMessages + " messages, and have " + factoidDao.count() + " factoids."));
        }
        return messages;
    }
}

File: src/main/kotlin/javabot/operations/StatsOperation.kt
package javabot.operations

import com.antwerkz.sofia.Sofia
import javabot.Javabot
import javabot.Message
import javabot.dao.AdminDao
import javabot.dao.FactoidDao
import java.time.Duration
import java.time.Instant.now
import javax.inject.Inject

class StatsOperation @Inject constructor(bot: Javabot, adminDao: AdminDao, var factoidDao: FactoidDao) : BotOperation(bot, adminDao) {
    private var numberOfMessages = 0

    override fun handleMessage(event: Message): List<Message> {
        val responses = arrayListOf<Message>()
        numberOfMessages++
        val message = event.value
        if ("stats".equals(message, ignoreCase = true)) {
            responses.add(Message(event, Sofia.botStats(Duration.between(startTime, now()).toDays(), numberOfMessages, factoidDao.count())))
        }
        return responses
    }

    companion object {
        private val startTime = now()
    }
}